packages/react-reconciler/src/ReactUpdateQueue.js

```diff
@@ -410,6 +410,16 @@ function getStateFromUpdate<State>(
           }
         }
         partialState = payload.call(instance, prevState, nextProps);
+        if (__DEV__ && workInProgress.mode & StrictMode) {
+          if (partialState === undefined) {
+            warning(
+              false,
+              'A setState updater function was called that returned undefined. ' +
```

Comment:
I stumbled a bit trying to remember what an updater function was. What do you think of:

> setState was called with an updater function that returned undefined

Do you think it would be worth adding an example, like `this.setState(() => undefined)`, or would that just be confusing?

Reply:
@sw-yx I'm a fan of simply dropping `updater`

Reply:
I'm not convinced we should drop `updater`, especially if we include a link to the docs.

@sw-yx , would the changes in this PR also cause a warning for:

```javascript
let newState;
// OOPS, forgot to assign newState
setState(newState); // newState === undefined
```